### PR TITLE
[dylib] New port

### PR DESCRIPTION
**Pull request by martin-olivier**
- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [X] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [X] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [X] The versioning scheme in `vcpkg.json` matches what upstream says.
- [X] The license declaration in `vcpkg.json` matches what upstream says.
- [X] The installed as the "copyright" file matches what upstream says.
- [X] The source code of the component installed comes from an authoritative source.
- [X] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is in the new port's versions file.
- [X] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by martin-olivier** [comment]
@microsoft-github-policy-service agree

File: ports/dylib/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    HEAD_REF main
+)
+
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}"
+)
+
+vcpkg_cmake_install()
+
+vcpkg_cmake_config_fixup(PACKAGE_NAME dylib CONFIG_PATH lib/cmake/dylib)
```
Comment (JonLiu1993) [inline ports/dylib/portfile.cmake:15]:
```suggestion
vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/dylib)
```

File: ports/dylib/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+vcpkg_fixup_pkgconfig()
+
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug" "${CURRENT_PACKAGES_DIR}/lib")
```
Comment (JonLiu1993) [inline ports/dylib/portfile.cmake:19]:
```suggestion
file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/lib")
```

File: ports/dylib/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE")
```
Comment (JonLiu1993) [inline ports/dylib/portfile.cmake:21]:
```suggestion
vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE")

```

**Comment by martin-olivier** [comment]
Thanks for your review @JonLiu1993 

I have an issue with the installation of the port.

> vcpkg install dylib
```
[...]
Elapsed time to handle dylib:x64-windows: 7.5 s
Total install time: 8.3 s
dylib is header-only and can be used from CMake via:

    find_path(DYLIB_INCLUDE_DIRS "dylib.hpp")
    target_include_directories(main PRIVATE ${DYLIB_INCLUDE_DIRS})
```

Then, when building a project that uses `dylib`:
```
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
C:/Users/molivier/Downloads/dylib/example/DYLIB_INCLUDE_DIRS
   used as include directory in directory C:/Users/molivier/Downloads/dylib/example
```

I expect something like `cxxopts` since it's a single file header only library like `dylib`
```
Elapsed time to handle cxxopts:x64-windows: 6.7 s
Total install time: 7.4 s
cxxopts provides CMake targets:

  # this is heuristically generated, and may not be correct
  find_package(cxxopts CONFIG REQUIRED)
  target_link_libraries(main PRIVATE cxxopts::cxxopts)
```

**Comment by JonLiu1993** [comment]
@martin-olivier, I tested the usage on my machine by your PR and I tested successfully.
CMakeLists.txt:
````
cmake_minimum_required (VERSION 3.8)

project(test)

find_path(DYLIB_INCLUDE_DIRS "dylib.hpp")
# Add source to this project's executable.
add_executable (test "usageTest.cpp")

target_include_directories(test PRIVATE ${DYLIB_INCLUDE_DIRS})
````
usageTest.cpp:
````
#include <stdio.h>
#include <dylib.hpp>

int main()
{
    printf("%s\n", "HELLO WORLD");
    return 0;
}
````
config:
````
1> CMake generation started for configuration: 'x64-Debug'.
1> Command line: "C:\WINDOWS\system32\cmd.exe" /c "%SYSTEMROOT%\System32\chcp.com 65001 >NUL && "C:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\COMMUNITY\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\CMAKE\CMake\bin\cmake.exe"  -G "Visual Studio 17 2022" -A x64  -DCMAKE_CONFIGURATION_TYPES:STRING="Debug" -DCMAKE_INSTALL_PREFIX:PATH="C:\Users\v-test\source\repos\usageTest\out\install\x64-Debug" -DCMAKE_TOOLCHAIN_FILE=F:/Feature-test/dylib/vcpkg/scripts/buildsystems/vcpkg.cmake "C:\Users\v-test\source\repos\usageTest" 2>&1"
1> Working directory: C:\Users\v-test\source\repos\usageTest\out\build\x64-Debug
1> [CMake] -- Selecting Windows SDK version 10.0.22621.0 to target Windows 10.0.19045.
1> [CMake] -- The C compiler identification is MSVC 19.36.32538.0
1> [CMake] -- The CXX compiler identification is MSVC 19.36.32538.0
1> [CMake] -- Detecting C compiler ABI info
1> [CMake] -- Detecting C compiler ABI info - done
1> [CMake] -- Check for working C compiler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.36.32532/bin/Hostx64/x64/cl.exe - skipped
1> [CMake] -- Detecting C compile features
1> [CMake] -- Detecting C compile features - done
1> [CMake] -- Detecting CXX compiler ABI info
1> [CMake] -- Detecting CXX compiler ABI info - done
1> [CMake] -- Check for working CXX compiler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.36.32532/bin/Hostx64/x64/cl.exe - skipped
1> [CMake] -- Detecting CXX compile features
1> [CMake] -- Detecting CXX compile features - done
1> [CMake] -- Configuring done (12.7s)
1> [CMake] -- Generating done (0.0s)
1> [CMake] -- Build files have been written to: C:/Users/v-test/source/repos/usageTest/out/build/x64-debug
1> Extracted CMake variables.
1> Extracted source files and headers.
1> Extracted code model.
1> Extracted toolchain configurations.
1> Extracted includes paths.
1> CMake generation finished.

````
build:
````
>------ Build All started: Project: usageTest, Configuration: x64-Debug ------
  MSBuild version 17.6.3+07e294721 for .NET Framework
  
    Checking Build System
    Building Custom Rule C:/Users/v-test/source/repos/usageTest/usageTest/CMakeLists.txt
    usageTest.cpp
    test.vcxproj -> C:\Users\v-test\source\repos\usageTest\out\build\x64-debug\usageTest\Debug\test.exe
    Building Custom Rule C:/Users/v-test/source/repos/usageTest/CMakeLists.txt

Build All succeeded.
````

**Comment by martin-olivier** [comment]
Hello @JonLiu1993,

Sorry, I forgot to specify the `CMAKE_TOOLCHAIN_FILE` for my previous build.

I have re-tested it and it works well.

**Review by JonLiu1993 — APPROVED** [review]

**Review by data-queue — APPROVED** [review]

**Pull request merged**